### ISSUE TITLE
Use dynamic attributeBindings to avoid manually specifying every possible attribute.

### ISSUE THREAD

**Issue by rwjblue** [DockYard/ember-one-way-controls#21]
Specifying each DOM attribute that any possible consumer will want is pretty annoying.  See [here for an example](https://github.com/dockyard/ember-one-way-controls/blob/515c01a7eb6498b47915514e235f4ce0088a6995/addon/components/one-way-input.js#L12-L37), ugh :crying_cat_face:.  This also has the negative impact of doing extra work when not needed (for supporting attributes that will never be used).

We could however, dynamically update `attributeBindings` in order to allow any attribute without us having to list them all.

That might look like this:

```js
// a blacklist for attributes that we are asking for folks to pass us in `attrs` (like `update` or `_sanitizeInput` functions for {{one-way-input}})
const attributeBlacklist = {
  update: true,
  _sanitizeInput: true
};

export default Ember.Component.extend({
  init() {
    this._super(...arguments);

    for (let key in this.attrs) {
        if (!attributeBlacklist[key]) {
          this.attributeBindings.push(key);  
        }
      }
  }
});
```

Demo Twiddle: https://ember-twiddle.com/dd0dd9b48140531ccccb

**Comment by poteto** [comment]
This is great! Want me to open a PR for it?

**Comment by bcardarella** [comment]
Does adding more attrBindings have any perf impact?

**Comment by rwjblue** [comment]
If you have time, yeah.  I likely won't have a chance before mid-week. 

This has been bugging me (having to add every attribute under the sun), and randomly this solution came to me this morning so I figured I'd share....

**Comment by rwjblue** [comment]
@bcardarella - This actually results in a significant reduction in the number of attributeBindings. If you checkout the link in the issue description you will see we have 15ish attributeBindings now, and after changing to use this pattern you would only have bindings for the attributes you actually used (which is pretty likely to be many fewer than we are using now).

**Comment by grapho** [comment]
just devil's advocate.. what would happen if a user puts in an arbitrary attr that is niether in the blacklist, nor in the acceptable list of input attributes?

maybe someone want an arbitrary `data-*` attr.. or just some random typo even..

would anything bad happen?

**Comment by rwjblue** [comment]
No, not to my knowledge. It would be the same as if you had typed that in HTML yourself...

**Comment by grapho** [comment]
ok that is good, i will allow this

**Comment by martndemus** [comment]
Closed because #67 and #68 are merged in.
